dspace-api/src/main/java/org/dspace/app/bulkedit/MetadataExportSearch.java

```diff
@@ -139,11 +139,13 @@ public void internalRun() throws Exception {
         DiscoverQuery discoverQuery =
             queryBuilder.buildQuery(context, dso, discoveryConfiguration, query, queryBuilderSearchFilters,
             "Item", 10, Long.getLong("0"), null, SortOption.DESCENDING);
+        // add configured limit
+        discoverQuery.setMaxResults(metadataDSpaceCsvExportService.getCsvExportLimit());
```

Comment:
This does not limit the size of the export. Here, `maxResults` is used to define the number of records per page that are retrieved with the iterator used to fetch the search results. It is better to remove this change and limit the number of records in the [loop that generates the results](https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/content/MetadataDSpaceCsvExportServiceImpl.java#L88-L92)